### PR TITLE
adds draw status game end condition

### PR DESCRIPTION
**Pull request by Dabble362**
the code is in the render() 

:0

### CODE REVIEW

File: src/components/Game.js [hunk 1]
```diff
@@ -76,6 +76,8 @@ export class Game extends React.Component {
     let status;
     if (winner) {
       status = "Winner: " + winner.player;
+    } else if (!current.squares.includes(null)) {
+      status = "The game is a draw!"
```
Comment (jduffey) [inline src/components/Game.js]: Add a semicolon at the end of this statement.

**Comment by jduffey** [comment]
This is a light suggestion but it's usually considered best practice for maintainability to declare your vars both in the order that they are used and only at the last moment they're needed. The (refactored) following code block has the logic for setting `status` moved above the declaration for `moves`.

```
render() {
  const history = this.state.history;
  const current = history[this.state.stepNumber];
  const winner = calculateWinner(current.squares);

  let status;
  if (winner) {
    status = "Winner: " + winner.player;
  } else if (!current.squares.includes(null)) {
    status = "The game is a draw!"
  } else {
    status = "Next player: " + (this.state.xIsNext ? "X" : "O");
  }

  const moves = history.map((step, move) => {
    const desc = move ? "Go to move #" + move : "Go to game start";
    return (
      <li key={move}>
        <button onClick={() => this.jumpTo(move)}>
          {move === this.state.stepNumber ? <b>{desc}</b> : desc}
        </button>{" "}
        {move > 0 && (
          <span>
            Player {move % 2 === 0 ? "O" : "X"}:
            {(history[move].location % 3) + 1},
            {Math.ceil((history[move].location + 1) / 3)}
          </span>
        )}
      </li>
    );
  });

// ...
```
If for whatever reason you were to want the `status` logic block to be after the `moves` block, then the following would be more conventional. Note that `current` and `winner` are declared only immediately before they're needed, and not before `moves`, where they aren't (this code block is a better example of what I'm trying to communicate here..):
```
render() {
  const history = this.state.history;

  const moves = history.map((step, move) => {
    const desc = move ? "Go to move #" + move : "Go to game start";
    return (
      <li key={move}>
        <button onClick={() => this.jumpTo(move)}>
          {move === this.state.stepNumber ? <b>{desc}</b> : desc}
        </button>{" "}
        {move > 0 && (
          <span>
            Player {move % 2 === 0 ? "O" : "X"}:
            {(history[move].location % 3) + 1},
            {Math.ceil((history[move].location + 1) / 3)}
          </span>
        )}
      </li>
    );
  });

  const current = history[this.state.stepNumber];
  const winner = calculateWinner(current.squares);
  
  let status;
  if (winner) {
    status = "Winner: " + winner.player;
  } else if (!current.squares.includes(null)) {
    status = "The game is a draw!"
  } else {
    status = "Next player: " + (this.state.xIsNext ? "X" : "O");
  }

// ...
```

In the return statement for this function, `status` is used before `moves` (because that's what you want the app to look like), so the first code block example above would be more appropriate. If hypothetically you wanted the app to look different and changed the render function such that `status` was used after `moves`, then you'd want to go with the second code block in order to satisfy the first rule of this convention ("declare your vars in the order that they are used").

**Review by jduffey — CHANGES_REQUESTED** [review]
Requesting a fix to one line (add the semicolon). Other than that, looks good!

**Comment by Dabble362** [comment]
> This is a light suggestion but it's usually considered best practice for maintainability to declare your vars both in the order that they are used and only at the last moment they're needed. The (refactored) following code block has the logic for setting `status` moved above the declaration for `moves`.
> 
> ```
> render() {
>   const history = this.state.history;
>   const current = history[this.state.stepNumber];
>   const winner = calculateWinner(current.squares);
> 
>   let status;
>   if (winner) {
>     status = "Winner: " + winner.player;
>   } else if (!current.squares.includes(null)) {
>     status = "The game is a draw!"
>   } else {
>     status = "Next player: " + (this.state.xIsNext ? "X" : "O");
>   }
> 
>   const moves = history.map((step, move) => {
>     const desc = move ? "Go to move #" + move : "Go to game start";
>     return (
>       <li key={move}>
>         <button onClick={() => this.jumpTo(move)}>
>           {move === this.state.stepNumber ? <b>{desc}</b> : desc}
>         </button>{" "}
>         {move > 0 && (
>           <span>
>             Player {move % 2 === 0 ? "O" : "X"}:
>             {(history[move].location % 3) + 1},
>             {Math.ceil((history[move].location + 1) / 3)}
>           </span>
>         )}
>       </li>
>     );
>   });
> 
> // ...
> ```
> 
> If for whatever reason you were to want the `status` logic block to be after the `moves` block, then the following would be more conventional. Note that `current` and `winner` are declared only immediately before they're needed, and not before `moves`, where they aren't (this code block is a better example of what I'm trying to communicate here..):
> 
> ```
> render() {
>   const history = this.state.history;
> 
>   const moves = history.map((step, move) => {
>     const desc = move ? "Go to move #" + move : "Go to game start";
>     return (
>       <li key={move}>
>         <button onClick={() => this.jumpTo(move)}>
>           {move === this.state.stepNumber ? <b>{desc}</b> : desc}
>         </button>{" "}
>         {move > 0 && (
>           <span>
>             Player {move % 2 === 0 ? "O" : "X"}:
>             {(history[move].location % 3) + 1},
>             {Math.ceil((history[move].location + 1) / 3)}
>           </span>
>         )}
>       </li>
>     );
>   });
> 
>   const current = history[this.state.stepNumber];
>   const winner = calculateWinner(current.squares);
>   
>   let status;
>   if (winner) {
>     status = "Winner: " + winner.player;
>   } else if (!current.squares.includes(null)) {
>     status = "The game is a draw!"
>   } else {
>     status = "Next player: " + (this.state.xIsNext ? "X" : "O");
>   }
> 
> // ...
> ```
> 
> In the return statement for this function, `status` is used before `moves` (because that's what you want the app to look like), so the first code block example above would be more appropriate. If hypothetically you wanted the app to look different and changed the render function such that `status` was used after `moves`, then you'd want to go with the second code block in order to satisfy the first rule of this convention ("declare your vars in the order that they are used").

Thank you Jed! This is very helpful!  I remember receiving this advice before and I would prefer to conform to these best practices.  It has been changed.

File: src/components/Game.js (hunk 1 above)
Comment (Dabble362) [reply src/components/Game.js]: Added. Thank you.

**Comment by jduffey** [comment]
Your most recent commit `6b7cca4` broke the app 🙂. Fire it up and look in the console for the reason why.

**Comment by jduffey** [comment]
> Your most recent commit `6b7cca4` broke the app 🙂. Fire it up and look in the console for the reason why.

@Dabble362 ^^

**Review by jduffey — APPROVED** [review]

**Pull request merged**